From f45160dfd901824b898751383bd299ee3ad4abdf Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Mon, 28 Jan 2013 15:15:23 +0100
Subject: [PATCH] (F18) Revert "udev: set device node permissions only at "add"
 events"

This reverts commit 48a849ee17fb25e0001bfcc0f28a4aa633d016a1.

Kay says a proper fix for the BZ is non-trivial, so the change that
broke 12-vdsm-lvm.rules should be reverted in F18.

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=903716
---
 src/udev/udev-node.c | 29 ++++++++++++++---------------
 1 file changed, 14 insertions(+), 15 deletions(-)

diff --git a/src/udev/udev-node.c b/src/udev/udev-node.c
index 1e378ad..7774303 100644
--- a/src/udev/udev-node.c
+++ b/src/udev/udev-node.c
@@ -279,23 +279,22 @@ static int node_fixup(struct udev_device *dev, mode_t mode, uid_t uid, gid_t gid
                 goto out;
         }
 
+        if ((stats.st_mode & 0777) != (mode & 0777) || stats.st_uid != uid || stats.st_gid != gid) {
+                log_debug("set permissions %s, %#o, uid=%u, gid=%u\n", devnode, mode, uid, gid);
+                chmod(devnode, mode);
+                chown(devnode, uid, gid);
+        } else {
+                log_debug("preserve permissions %s, %#o, uid=%u, gid=%u\n", devnode, mode, uid, gid);
+        }
+
         /*
-         * Set permissions and selinux file context only on add events. We always
-         * set it on bootup (coldplug) with "trigger --action=add" for all devices
-         * and for any newly added devices (hotplug). We don't want to change it
-         * later, in case something else has applied custom settings in the meantime.
+         * Set initial selinux file context only on add events.
+         * We set the proper context on bootup (triger) or for newly
+         * added devices, but we don't change it later, in case
+         * something else has set a custom context in the meantime.
          */
-        if (strcmp(udev_device_get_action(dev), "add") == 0) {
-                if ((stats.st_mode & 0777) != (mode & 0777) || stats.st_uid != uid || stats.st_gid != gid) {
-                        log_debug("set permissions %s, %#o, uid=%u, gid=%u\n", devnode, mode, uid, gid);
-                        chmod(devnode, mode);
-                        chown(devnode, uid, gid);
-                } else {
-                        log_debug("preserve permissions %s, %#o, uid=%u, gid=%u\n", devnode, mode, uid, gid);
-                }
-
-                label_fix(devnode, true, false);
-        }
+        if (strcmp(udev_device_get_action(dev), "add") == 0)
+            label_fix(devnode, true, false);
 
         /* always update timestamp when we re-use the node, like on media change events */
         utimensat(AT_FDCWD, devnode, NULL, 0);
